wormhole-connect/src/hooks/useFetchTransactionHistory.ts

```diff
@@ -0,0 +1,243 @@
+    //   There has been unexpected conversaion rates in API responses. Therefore we are
+    //   sniffing out the decimals from the conversion rate of between these two values.
+    const conversionRate =
+      Number(standarizedProperties.amount) / Number(data.tokenAmount);
+    const decimals = conversionRate.toString().length - 1;
+
+    const sentAmountDisplay = sdkAmount.display(
+      {
+        amount: standarizedProperties.amount,
+        decimals,
+      },
+      0,
+    );
+
+    const receiveAmountValue = Math.max(
+      standarizedProperties.amount - standarizedProperties.fee,
+      0,
+    );
+    const receiveAmountDisplay = sdkAmount.display(
+      {
+        amount: receiveAmountValue.toString(),
+        decimals,
+      },
+      0,
+    );
+
+    const feeAmountDisplay = sdkAmount.display(
+      {
+        amount: standarizedProperties.fee,
+        decimals,
+      },
+      0,
+    );
+
+    const txData: Transaction = {
+      txHash: sourceChain.transaction?.txHash,
+      sender: standarizedProperties.fromAddress || sourceChain.from,
+      amount: sentAmountDisplay,
+      amountUsd: data.usdAmount,
+      recipient: standarizedProperties.toAddress,
+      toChain,
+      fromChain,
+      tokenKey: tokenConfig.key,
+      receivedTokenKey,
+      receiveAmount: receiveAmountDisplay,
+      relayerFee: {
+        fee: Number(feeAmountDisplay),
+        tokenKey: toChainConfig?.gasToken || '',
+      },
+      tokenAddress: standarizedProperties.tokenAddress,
+      senderTimestamp: sourceChain?.timestamp,
+      receiverTimestamp: targetChain?.timestamp,
+    };
+
+    return txData;
+  };
+
+  const SUPPORTED_APPIDS = {
+    PORTAL_TOKEN_BRIDGE: parseTokenBridgeTx,
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const parseTransactions = useCallback((allTxs: Array<any>) => {
+    const parsedTxs: Array<Transaction> = allTxs.map((tx) => {
+      const appIds = tx.content?.standarizedProperties?.appIds || [];
+
+      let txParser;
+
+      appIds.forEach((appId) => {
+        if (typeof SUPPORTED_APPIDS[appId] === 'function') {
+          txParser = SUPPORTED_APPIDS[appId];
+        }
+      });
+
+      return txParser?.(tx);
+    });
+
+    return parsedTxs.filter((tx) => !!tx);
+  }, []);
+
+  useEffect(() => {
+    let cancelled = false;
+
+    const headers = new Headers({
+      accept: 'application/json',
+    });
+
+    const fetchTransactions = async () => {
+      setIsFetching(true);
+      let data;
+
+      try {
+        const res = await fetch(
+          `${config.wormholeApi}api/v1/operations?address=${sendingWallet.address}&page=${page}&pageSize=${pageSize}`,
+          { headers },
+        );
+
+        data = await res.json();
+      } catch (error) {
+        if (!cancelled) {
+          setError(`Error fetching transaction history: ${error}`);
+        }
+
+        setIsFetching(false);
+      }
+
+      if (!cancelled) {
```

Comment:
IMO it's confusing how this block runs even when we caught an error above. can you return early in the `catch` or maybe put this under the `try`?

If you put this under the `try`, you could call `setIsFetching(false);` for both cases by putting it inside a `finally`

```ts
try {
  // happy path
} catch (e) {
  // sad path
} finally {
  setIsFetching(false);
}
```

Reply:
If we caught and error inside try/catch, data will be falsy and this block won't run. I wanted like to keep try/catch minimal to the things that we actually want to monitor. That being said I can apply the recommended pattern to increase readability.

Reply:
yeah data being falsy is what I'm saying makes this a little hard to read. I'm not hung up on this one, just a comment. you can leave it as is if you prefer.